scipy/signal/windows.py

```diff
@@ -693,8 +745,8 @@ def hann(M, sym=True):
         return np.ones(M)
     M, needs_trunc = _extend(M, sym)
 
-    n = np.arange(0, M)
-    w = 0.5 - 0.5 * np.cos(2.0 * np.pi * n / (M - 1))
+    a = [0.5, 0.5]
+    w = _cos_win(M, a)
```

Comment:
ok I shortened most of them except one that was too long for a single line that way.

could also put the guards and stuff into the _cos_win function so that blackman becomes just 

    return _cos_win(M, a=[0.42, 0.50, 0.08], sym)

for instance.